src/wallet/feebumper.cpp

```diff
@@ -223,8 +230,9 @@ Result CreateRateBumpTransaction(CWallet* wallet, const uint256& txid, const CCo
 
     // Fill in recipients(and preserve a single change key if there is one)
     std::vector<CRecipient> recipients;
+    const auto* change_output = reduce_output > -1 && (size_t)reduce_output < wtx.tx->vout.size() ? &wtx.tx->vout[reduce_output] : nullptr;
     for (const auto& output : wtx.tx->vout) {
-        if (!wallet->IsChange(output)) {
+        if (change_output ? output != *change_output : !wallet->IsChange(output)) {
```

Comment:
Couldn't this result in the `reduce_output` getting *increased*? That could be a serious problem in some cases...

Reply:
I'm not sure I follow. In which case would `reduce_output` increase?